### DISCUSSION TITLE
Help me understand Remote Extension Host builds and how to use them, please

### DISCUSSION THREAD

**Discussion by HoboDev** [VSCodium/vscodium#1945] [Q&A]
I'm trying to understand what I can do with the ```reh``` builds. (and how)
In some comments I've seen they are for servers. Also this repo needs them installed manual for some BSDs: https://github.com/jeanp413/open-remote-ssh
And specific builds like the ```alpine-reh``` is mentioned in the issues.

I've looked through the documentation, the website and have been searching for about 3 hours...
So, what exactly does this do? How does this work with open-remote-ssh?
Is this some kind of devcontainer-esque solution?
Where can I find the documentation for that?
How do I use it?

Any help is very much appreciated.

**Comment by JohnVeness** [comment]
Came here to ask exactly this. The "reh" assets are listed in the GitHub releases section, but I couldn't find any explanation as to what they are?

**Reply by daiyam** [reply]
Yep, more docs are needed

- Remote Host (`reh`) is the server component for remote ssh/wsl which runs it on a "remote" computer and makes that remote computer accessible via VSCodium.
- Web Host (`reh-web`) is the server component of the command `codium serve-web` which runs it locally and makes VSCodium accessible via a browser.

Is good enough?

**Reply by xBZZZZ** [reply]
> makes that remote computer accessible via VSCodium

how to use vscodium as client that connects to `reh` server?

**Reply by daiyam** [reply]
Use https://open-vsx.org/extension/jeanp413/open-remote-ssh

**Reply by xBZZZZ** [reply]
> Use https://open-vsx.org/extension/jeanp413/open-remote-ssh

why is this separate extension and what does it have to do with `SSH`?

does remote thing not work like:
1. run `bin/codium-server --host=172.16.133.129 --port=3464` inside computer 1
2. somehow enter host=`172.16.133.129` port=`3464` inside vscodium running on computer 2
3. edit computer 1 files using vscodium running on computer 2

?

**Reply by daiyam** [reply]
`bin/codium-server` isn't securized so SSH adds a layer of auth and security.

But, yeah, you could do as you described but I haven't tested it.

**Reply by nick-walt** [reply]
> Yep, more docs are needed
> 
Thanks, yeah -- even if the documentation refers the reader to Microsoft's documentation for setting up the remote environment -- assuming that the OVSX version "open-remote-ssh" process to setup and connect remotely is the same as documented by Microsoft, is that correct?

> * Remote Host (`reh`) is the server component for remote ssh/wsl which runs it on a "remote" computer and makes that remote computer accessible via VSCodium.
>
Does this mean that, unlike the automatic upload and installation of the server component performed by VSCode, users of VSCodium must manually upload the server component to a specific location? Is that location documented anywhere on the VSCodium site?

> * Web Host (`reh-web`) is the server component of the command `codium serve-web` which runs it locally and makes VSCodium accessible via a browser.
>
Is it possible to run the Web Host on a VM and load it in a browser? If so this will provide strong isolation of the dev environment. I'm running Gnome 48 and it looks like VSCodium and Code are not fully compatible with Wayland or xWayland. Dialog boxes do not show content and a mouse-over action is required to reveal the buttons in order to Cancel the dialong, so running in a browser might solve this problem until the editors work well on Wayland.

Cheers

**Reply by daiyam** [reply]
> > * Remote Host (`reh`) is the server component for remote ssh/wsl which runs it on a "remote" computer and makes that remote computer accessible via VSCodium.
> 
> Does this mean that, unlike the automatic upload and installation of the server component performed by VSCode, users of VSCodium must manually upload the server component to a specific location? Is that location documented anywhere on the VSCodium site?

With `open-remote-ssh` extension, the server component is automatically installed when you connect to the remoter server. It's the extension whose manage that.

> > * Web Host (`reh-web`) is the server component of the command `codium serve-web` which runs it locally and makes VSCodium accessible via a browser.
> 
> Is it possible to run the Web Host on a VM and load it in a browser? If so this will provide strong isolation of the dev environment. I'm running Gnome 48 and it looks like VSCodium and Code are not fully compatible with Wayland or xWayland. Dialog boxes do not show content and a mouse-over action is required to reveal the buttons in order to Cancel the dialong, so running in a browser might solve this problem until the editors work well on Wayland.

Yes, use the `reh-web` archive. But do not expose it since it isn't secured.

**Reply by nick-walt** [reply]
Thanks, daiyam. What I found confusing was that the VSCodium downloads section includes a remote hosts server in their download links and I thought this was a manual process. Does `open-remote-ssh` download the same remote server that Microsoft's remote extension installs?

Do you know if the `open-remote-ssh` extension is a fork of Microsoft's remote extension?
`reh-web` will run in a local VM and I'll make sure to only allow connection from the host. Cheers.

**Reply by daiyam** [reply]
> Do you know if the `open-remote-ssh` extension is a fork of Microsoft's remote extension?

No. The MS remote extension is close-sourced.

>  Does `open-remote-ssh` download the same remote server that Microsoft's remote extension installs?

No. It uses VSCodium `reh`.

**Comment by nick-walt** [comment]
Do you know if the Snap package of VSCodium can be installed with a custom snap confinement mode such as `--strict` and then opened back up a little with individually applied Connections and Interfaces? I want to sandbox VSCodium as much as possible.

**Reply by daiyam** [reply]
No, it's the opposite. It has to been installed with the `--classic` flag...
